.ci/lib.sh

```diff
@@ -246,6 +246,31 @@ delete_stale_kata_resource()
 	done
 }
 
+delete_kata_repo_registrations()
+{
+	case "$ID" in
+		ubuntu)
+			local apt_file="/etc/apt/sources.list.d/kata-containers.list"
+			if [ -f "$apt_file" ]; then
+				info "Removing Kata apt file [$apt_file]"
+				sudo rm -f "$apt_file"
+			fi
+			# apt-key output format changed at ubuntu 16.10
+			if [ "$VERSION_ID" \< "16.10" ]; then
+				kata_uuid="$(sudo apt-key list | awk '$2=="home:katacontainers" {print prev} {prev=$2}')"
+			else
+				kata_uuid="$(sudo apt-key list | awk '$4=="home:katacontainers" {print prev} {prev=$0}')"
+			fi
+			if [ -n "$kata_uuid" ]; then
```

Comment:
I wonder if we want a `[ -z "$kata_uuid" ]` test in here so we get an alert when format changes yet again? :)

Reply:
it would be nice, but the only problem I see is if we run on a machine that has no kata key installed (like the first time we use an instance, or if we ever get to bare metal one-off deployments which I hope to achieve on day), then it will fail as the kata key won't be present.

Reply:
In which case how about we error if the key is installed (`apt-key list | grep home:katacontainers`) but `$kata_uuid` is blank?

Reply:
let me add that then..

Reply:
and... one more time....
 - predicated the key parse/remove on a kata key being present
 - fixed a bug - needed to remove the 2048R/ key frontage before handing back to apt-key del